### ISSUE TITLE
`Button` component re-renders incorrectly when dynamically updated from `contained` to `outlined`

### ISSUE THREAD

**Issue by fabriziocucci** [callstack/react-native-paper#3137]
### TL;DR

I'm quite sure I'm doing something wrong here but any advice would be much appreciated! 🙂 

### Current behaviour

When changing the `Button` mode dynamically (e.g. from `contained` to `outlined`), the component re-renders incorrectly (see the dark shadow in the screenshot below or try to randomly click the buttons in the Expo snack below).

### Expected behaviour

When changing the `Button` mode dynamically (e.g. from `contained` to `outlined`), the component should re-render correctly.

### Code sample

Expo snack: https://snack.expo.dev/@fabrizio.cucci/ab4d4d

### Screenshots (if applicable)

<img width="772" alt="Screenshot 2022-04-02 at 12 28 29" src="https://user-images.githubusercontent.com/8156463/161381064-7dd1234b-e698-4906-820e-11ff2799e494.png">

### Your Environment

| software              | version
| --------------------- | -------
| ios or android        | n/a
| react-native          | 0.64.3
| react-native-paper    | 4.12.0
| node                  | 16.13.0
| yarn           | 1.22.15
| expo              | 44.0.0

**Comment by drplauska** [comment]
Reproducible on Android. Seems to be a problem with elevation. To be exact, if we updated this file
`src/components/Button.tsx`
on line 275
to this:
`{...(theme.isV3 && { elevation: 0 })}`
the bug resolves, but this removes `elevation` at all
